### PR TITLE
Native view

### PR DESCRIPTION
**Pull request by anthonyccri**
WIP for comparison purposes.

### CODE REVIEW

File: geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        });
+    }
+
+    @Override
+    public IndexType indexType() {
+        return indexType;
+    }
+
+    public SimpleFeatureType buildSimpleFeatureType() {
+        return sft;
```
Comment (cne1x) [inline geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java]: Do we care that someone could modify this shared object externally, e.g, modify user data?

File: geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @Override
+    public SimpleFeature buildSimpleFeature(T t, Geometry geometry) {
```
Comment (cne1x) [inline geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java]: It doesn't look like we're using the ```t``` anywhere in this method (nor the other), but I'm probably just misunderstanding something.  Is there meant to be a slot for the native object in the feature type we build, use?

**Comment by jnh5y** [comment]
Failed build due to headers.

File: geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+import java.util.Date;
+import java.util.UUID;
+
+@InterfaceStability.Unstable
+public class DefaultSimpleFeatureView<T> implements SimpleFeatureView<T> {
+
+    private IndexType indexType;
+
+    private Logger logger = LoggerFactory.getLogger(DefaultSimpleFeatureView.class);
```
Comment (jahhulbert-ccri) [inline geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java]: make this ```static final``` as well

File: geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.UUID;
+
+@InterfaceStability.Unstable
+public class DefaultSimpleFeatureView<T> implements SimpleFeatureView<T> {
+
+    private IndexType indexType;
```
Comment (jahhulbert-ccri) [inline geomesa-native-api/src/main/java/org/locationtech/geomesa/api/DefaultSimpleFeatureView.java]: make ```indexType, sft, and builderPool``` final to enforce it is assigned by the constructor?

**Pull request closed without merging**